cmd/cp-main_test.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Minio Client (C) 2019 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+
+	"errors"
+	"reflect"
+	"testing"
+)
+
+func TestParseMetaData(t *testing.T) {
+	metaDataCases := []struct {
+		input  string
+		output map[string]string
+		err    error
+		status bool
+	}{
+		// success scenerio
+		{"key1=value1,key2=value2", map[string]string{"key1": "value1", "key2": "value2"}, nil, true},
+		// using different delimitter, other than ',' between multiple meta data
+		{"key1=value1;key2=value2", nil, errors.New("specified metadata should be of form key1=value1,key2=value2,... and so on"), false},
```

Comment:
we must be able to reuse `ErrInvalidMetadata` in the tests instead of `errors.New(..)`.

Reply:
done.